src/libcore/iter/iterator.rs

```diff
@@ -1242,7 +1242,7 @@ pub trait Iterator {
     /// // let's try that again
     /// let a = [1, 2, 3];
     ///
-    /// let mut iter = a.into_iter();
```

Comment:
Oh dear, these changes mean that this is technically a breaking change, right?

Reply:
If I understand correctly what's going on, this "breaking change" is no different from the way adding a new method to `Iterator` which (if e.g. itertools already implements that method) would break all users of itertools who used the method syntax instead of the UFC syntax to call said method.

Reply:
Correct yeah. This doesn't mean it's impossible to land, just that it needs more scrutiny/review.

Reply:
This does feel like one of those changes that, while (allowed) breaking, people have wanted for a very long time and we definitely want to do at some point.